fn/concurrency.go

```diff
@@ -32,3 +34,43 @@ func ParSlice[V any](ctx context.Context, s []V, f ErrFunc[V]) error {
 
 	return errGroup.Wait()
 }
+
+// ParSliceErrCollect can be used to execute a function on each element of a
+// slice in parallel. This function is fully blocking and will wait for all
+// goroutines to finish. Error will be collected and returned as a map of slice
+// element index to error. Active goroutines limited with number of CPU.
+func ParSliceErrCollect[V any](ctx context.Context, s []V,
+	f ErrFunc[V]) (map[int]error, error) {
+
+	errGroup, ctx := errgroup.WithContext(ctx)
+	errGroup.SetLimit(runtime.NumCPU())
+
+	var instanceErrorsMutex sync.Mutex
+	instanceErrors := make(map[int]error, len(s))
+
+	for idx, v := range s {
+		v := v
```

Comment:
nit: this is no longer required, as we're now using Go 1.22.

Reply:
I've gone with `s[idx]` in the latest commits. Are you saying that `v := v` isn't required because of Go 1.22? Why is that, can you say more please?

Reply:
https://go.dev/blog/loopvar-preview